_posts/2016-07-14-official-docker-image.md

```diff
@@ -0,0 +1,170 @@
+
+The redirection from port `80` to port `443` applies to all the common requests sent
+to Portus. The only requests that are **not** redirected are the ones sent to
+the `v2/webhooks/events` endpoint. This is the endpoint used by the Docker
+Registry to send notifications to Portus.
+
+Sending Registry's notification to the HTTPS endpoint would require Portus'
+certificate to be added to the Docker container running the registry. This is
+not hard to be done, but involves some changes to the official Registry image
+maintained by Docker Inc (or some hacks like a custom entrypoint added via a volume).
+
+### The init script
+
+As mentioned before, this Docker image has a init script which takes care of the
+following actions:
+
+  1. Setup the database required by Portus
+  2. Import all the `.crt` located under `/certificates`
+  3. Start Apache
+
+The next sections will provide more details about these steps.
+
+### Volumes
+
+Portus' state is stored inside of a MariaDB database. This makes this Docker
+image stateless.
+
+### External database
+
+This image has a custom `init` script that takes care of configuring the external
+database.
+
+The script will keep trying to reach the database for 90 seconds. A 5 seconds
+pause is done after each failed attempt. The container will exit with an error
+message if the database is not reachable.
+
+The script will take care of the creation of the Portus database and its initial
+population via the usual Rails procedures.
+
+The database is automatically migrated whenever a new migration is introduced
+by the upstream project.
+
+### Secrets and certificates
+
+Portus requires both a SSL key and a certificate to serve its contents over
+HTTPS.
+These files must be located here:
+
+  * SSL certificate file: `/certificates/portus-ca.crt`
+  * SSL certificate key file: `/secrets/portus.key`
+
+The same key is also used to sign the JWT tokens issued to authenticate all the
+docker clients against the Registry.
+
+It's also required to add the certificate of the Registry when the latter one
+uses TLS to secure itself.
+The Registry certificate must be placed inside of `/certificates`, the `init`
+script of this image will automatically import it if it ends with the `.crt`
+extension.
+
+### Logging
+
+Both Portus' and Apache's messages are redirected to `stdout` and `stderr`. This
+makes it possible to handle the logs of this image in the usual "Docker ways".
+
+## Deployment
+
+It's possible to deploy this image using one of the existing orchestration
+solutions for Docker images.
+
+However we recommend the usage of [kubernetes](http://kubernetes.io/). We will
+soon release a reference implementation.
+
+### Environment variables
+
+The following environment variables must be defined to run the image:
+
+Security related settings:
+
+  * `PORTUS_SECRET_KEY_BASE`: you can generate it using `openssl rand -hex 64`
```

Comment:
To me, this reads: `openssl rand -hex 64` will generate the `PORTUS_SECRET_KEY_BASE` env variable. That's not true, right? Perhaps the wording could be changed here and the line below.